### PR TITLE
Walk through offre folders then files in NetexIDF

### PR DESCRIPTION
**Pull request by woshilapin**

### CODE REVIEW

File: src/netex_idf/offre.rs [hunk 1]
```diff
@@ -0,0 +1,82 @@
+use crate::{model::Collections, Result};
+use failure::format_err;
+use log::{info, warn};
+use minidom::Element;
+use std::{fs::File, io::Read, path::Path};
+use walkdir::WalkDir;
+
+pub fn read_offre_folder<P: AsRef<Path>>(
```
Comment (datanel) [inline src/netex_idf/offre.rs]: if this function is only called by `read` function, is it necessary to have `P: AsRef<Path>` ?

File: src/netex_idf/read.rs [hunk 2]
```diff
@@ -45,6 +47,21 @@ where
...
+        .into_iter()
+        .filter_map(std::result::Result::ok)
+        .filter(|dir_entry| {
+            dir_entry
+                .metadata()
+                .map(|metadata| metadata.is_dir())
+                .unwrap_or_default()
+        })
+    {
+        info!("Reading offre in folder {:?}", offre_folder);
```
Comment (datanel) [inline src/netex_idf/read.rs]: for better log `Reading offre in folder DirEntry("input/OFFRE_TOTO")` --> Reading offre in folder input/OFFRE_TOTO`
```suggestion
        info!("Reading offre in folder {:?}", offre_folder.path());
```

File: src/netex_idf/offre.rs (hunk 1 above)
Comment (woshilapin) [reply src/netex_idf/offre.rs]: Yes, it's still a `pub` function but at the same time, it's not really exposed on `transit_model` (only internally). I removed it for the sake of simplicity of the code in 292355f35b9e1b51430d00f88e73e1be15b69ce9.

**Comment by woshilapin** [comment]
@ArnaudOggy As discussed together, I updated the PR by changing `offre` to the English version of it `offer`.

File: src/netex_idf/read.rs [hunk 3]
```diff
@@ -45,6 +47,21 @@ where
     stops::from_path(&path.join("arrets.xml"), &mut collections)?;
     // TODO : use _lines_netex_idf to get trips>physical_mode_id
     let _lines_netex_idf = lines::from_path(&path.join("lignes.xml"), &mut collections)?;
+    for offer_folder in WalkDir::new(path)
+        .min_depth(1)
+        .max_depth(1)
+        .into_iter()
+        .filter_map(std::result::Result::ok)
```
Comment (ArnaudOggy) [inline src/netex_idf/read.rs]: proposal
.filter_map(|entry| entry.ok())
.filter(|entry| entry.file_type().is_dir())

Comment (woshilapin) [reply src/netex_idf/read.rs]: Nice, I was not a fan of this `.metadata()` API but it seems I completely missed the `.file_type()` one. Fixed in 3ebad02c1150254f0a168007a0c4ab44fc255ca6.

File: src/netex_idf/offers.rs [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        let mut calendriers_file_content = String::new();
+        calendriers_file.read_to_string(&mut calendriers_file_content)?;
+        let calendriers: Element = calendriers_file_content
+            .parse()
+            .map_err(|_| format_err!("Failed to open {:?}", calendrier_path))?;
+        info!("Reading {:?}", calendrier_path);
+        parse_calendriers(&calendriers)?;
+    } else {
+        warn!(
+            "Offer {:?} ignored because it does not contain the 'calendrier.txt' file.",
```
Comment (ArnaudOggy) [inline src/netex_idf/offers.rs]: calendriers.xml and not .calendrier.txt.
You can use a variable to prevent that.

Comment (woshilapin) [reply src/netex_idf/offers.rs]: Done in 8ebcb63c268de7fdcaaf9462bce62c788235ba0a.

**Pull request merged**